core-web/libs/portlets/dot-locales/portlet/src/lib/dot-locales-list/components/dot-locale-create-edit/DotLocaleCreateEdit.component.ts

```diff
@@ -0,0 +1,138 @@
+import { BehaviorSubject } from 'rxjs';
+
+import { CdkCopyToClipboard } from '@angular/cdk/clipboard';
+import { CommonModule } from '@angular/common';
+import { ChangeDetectionStrategy, Component, inject, OnInit } from '@angular/core';
+import {
+    AbstractControl,
+    FormControl,
+    FormGroup,
+    ReactiveFormsModule,
+    Validators
+} from '@angular/forms';
+
+import { SelectItem } from 'primeng/api/selectitem';
+import { ButtonModule } from 'primeng/button';
+import { DropdownModule } from 'primeng/dropdown';
+import { DynamicDialogConfig, DynamicDialogRef } from 'primeng/dynamicdialog';
+import { InputTextModule } from 'primeng/inputtext';
+import { TooltipModule } from 'primeng/tooltip';
+
+import { DotMessageService } from '@dotcms/data-access';
+import { DotISOItem, DotLanguage } from '@dotcms/dotcms-models';
+import { DotMessagePipe } from '@dotcms/ui';
+
+export interface DotLocaleCreateEditData {
+    languages: DotISOItem[];
+    countries: DotISOItem[];
+    locale: DotLanguage | null;
+    localeList: DotLanguage[];
+}
+
+@Component({
+    selector: 'dot-locale-create-edit',
+    standalone: true,
+    imports: [
+        CommonModule,
+        DropdownModule,
+        ReactiveFormsModule,
+        ButtonModule,
+        DotMessagePipe,
+        InputTextModule,
+        CdkCopyToClipboard,
+        TooltipModule
+    ],
+    templateUrl: './DotLocaleCreateEdit.component.html',
+    styleUrl: './DotLocaleCreateEdit.component.scss',
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class DotLocaleCreateEditComponent implements OnInit {
+    readonly config: DynamicDialogConfig<DotLocaleCreateEditData> = inject(DynamicDialogConfig);
+    private readonly dotMessageService = inject(DotMessageService);
+
+    ref = inject(DynamicDialogRef);
+    languageId: AbstractControl<string, string> | null | undefined;
+    form: FormGroup = new FormGroup({});
+    data: DotLocaleCreateEditData = this.config.data || {
+        languages: [],
+        countries: [],
+        locale: null,
+        localeList: []
+    };
+    showError = new BehaviorSubject<boolean>(false);
+    localeType: SelectItem[] = [
+        { label: this.dotMessageService.get('locale.standard.locale'), value: 1 },
+        { label: this.dotMessageService.get('locale.Custom.locale'), value: 2 }
+    ];
+
+    ngOnInit(): void {
+        this.initForm();
+    }
+
+    handleSubmit(): void {
+        if (this.form.valid) {
+            const { language, languageCode, country, countryCode } = this.form.value;
+            const isDuplicate = this.data.localeList.some(
+                (locale) => locale.isoCode?.toLowerCase() === this.getISOCode()
+            );
+
+            if (this.data.locale?.id || (!this.data.locale?.id && !isDuplicate)) {
```

Comment:
I think this is the same as `this.data.locale?.id || !isDuplicate`

Reply:
not really.. moving those conditions to functions to be more descriptive.